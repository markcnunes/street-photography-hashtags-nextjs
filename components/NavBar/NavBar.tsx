import React from 'react';
import styled from '@emotion/styled';

import ActiveLink from 'components/ActiveLink/index';

const StyledNavbar = styled('nav')(({ theme }) => ({
  display: 'inline-flex',
  alignItems: 'center',
  padding: 20,
  border: 0,
  outline: 0,
  cursor: 'pointer',
  borderRadius: 5,
  textDecoration: 'none',
  a: {
    textDecoration: 'none',
    color: theme.colors.white,
    padding: 10,
    border: '1px solid transparent',
    borderRadius: theme.radii[2],
    '&.is-active': {
      borderColor: theme.colors.muted,
    },
    '&:hover': {
      borderColor: 'currentColor',
    },
    '& + a': {
      marginLeft: 10,
    },
  },
}));

const Navbar = () => {
  return (
    <StyledNavbar role='navigation' aria-label='main navigation'>
      <ActiveLink activeClassName='is-active' href='/'>
        <a>Categories</a>
      </ActiveLink>

      <ActiveLink activeClassName='is-active' href='/about'>
        <a className='navbar-item'>About</a>
      </ActiveLink>

      <ActiveLink activeClassName='is-active' href='/contribution'>
        <a className='navbar-item'>Contribution</a>
      </ActiveLink>
    </StyledNavbar>
  );
};

export default Navbar;
