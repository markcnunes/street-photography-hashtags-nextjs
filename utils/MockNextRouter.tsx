import { RouterContext } from 'next/dist/next-server/lib/router-context';
import { NextRouter } from 'next/router';
import { ReactNode } from 'react';

interface MockNextRouterProps extends Partial<NextRouter> {
  children?: ReactNode;
}

/**
 * Override the default NextRouter render with different attributes.
 * e.g. change the path or locale for tests.
 */

const MockNextRouter = ({ children, ...props }: MockNextRouterProps) => {
  const mockRouter: NextRouter = {
    route: '/',
    pathname: '',
    query: {},
    asPath: '/',
    basePath: '',
    locale: 'en',
    locales: [''],
    defaultLocale: 'en',
    // domainLocales?: DomainLocales,
    isLocaleDomain: true,
    push: jest.fn(() => Promise.resolve(true)),
    replace: jest.fn(() => Promise.resolve(true)),
    reload: jest.fn(() => Promise.resolve(true)),
    back: jest.fn(() => Promise.resolve(true)),
    prefetch: jest.fn(),
    beforePopState: jest.fn(() => Promise.resolve(true)),
    events: {
      on: jest.fn(),
      off: jest.fn(),
      emit: jest.fn(),
    },
    isFallback: false,
    isReady: true,
    isPreview: false,
  };

  return (
    <RouterContext.Provider value={{ ...mockRouter, ...props }}>
      {children}
    </RouterContext.Provider>
  );
};

export default MockNextRouter;
