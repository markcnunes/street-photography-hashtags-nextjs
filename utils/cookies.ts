// import ReactGA from 'react-ga';

import GA4React from 'ga-4-react';

export const cookieConsent = 'sphCookieconsent';

/*
 * Check if cookie consentment is true before initializing and rendering it
 */
export const getCookie = (key: string): string | undefined => {
  var b = document.cookie.match('(^|;)\\s*' + key + '\\s*=\\s*([^;]+)');
  return b ? b.pop() : '';
};

export const gaCode = 'G-W7QKGLQ9YV';
const initializeCookies = () => {
  if (!window.location.href.includes('localhost')) {
    const ga4react = new GA4React(gaCode);
    ga4react.initialize();
    // reload page to get the track running
    window.location.reload();
  }
};

export const renderCookies = (): any => {
  getCookie(cookieConsent) === 'true' && initializeCookies();
};

export default initializeCookies;
