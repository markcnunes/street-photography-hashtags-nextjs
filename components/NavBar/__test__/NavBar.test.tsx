import React from 'react';
import MockNextRouter from 'utils/MockNextRouter';
import { render } from 'utils/test-utils';
import NavBar from '../NavBar';

describe('<NavBar />', () => {
  it('matchs snapshot', () => {
    const { asFragment } = render(
      <MockNextRouter>
        <NavBar />
      </MockNextRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
