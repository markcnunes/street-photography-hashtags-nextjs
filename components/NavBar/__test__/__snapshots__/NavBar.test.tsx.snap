// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> matchs snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
  border: 0;
  outline: 0;
  cursor: pointer;
  border-radius: 5px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
  padding: 10px;
  border: 1px solid transparent;
  border-radius: 4px;
}

.emotion-0 a.is-active {
  border-color: #536075;
}

.emotion-0 a:hover {
  border-color: currentColor;
}

.emotion-0 a+a {
  margin-left: 10px;
}

<nav
    aria-label="main navigation"
    class="emotion-0 emotion-1"
    role="navigation"
  >
    <a
      class="is-active"
      href="/"
    >
      Categories
    </a>
    <a
      class="navbar-item"
      href="/about"
    >
      About
    </a>
    <a
      class="navbar-item"
      href="/contribution"
    >
      Contribution
    </a>
  </nav>
</DocumentFragment>
`;
